examples/dataset/crossing_pedestrian_detection/crossing_pedestrian_detection/upload_results.py

```diff
@@ -0,0 +1,315 @@
+
+    return FrameMetrics(TP=tps, FP=fps, TN=tns, FN=fns)  # type: ignore
+
+
+def compute_metrics(dataset: str, inference_data: Dict[str, List[ScoredLabeledBoundingBox]]) -> pd.DataFrame:
+    dataset_df = download_dataset(dataset)
+    results = []
+    for row in dataset_df.itertuples():
+        filemapping = Path(row.locator.split("/")[-1]).stem
+        inference_bboxes = inference_data[filemapping]
+        frame_metrics_combined, df_raw_inferences = compute_frame_metrics_row(
+            row.high_risk,
+            row.high_risk_pids,
+            inference_bboxes,
+        )
+        all_pedestrian_by_frames = defaultdict(list)
+        for pid, frame_metrics in frame_metrics_combined.items():
+            for frame_id, metrics in frame_metrics.items():
+                all_pedestrian_by_frames[frame_id].append(metrics)
+
+        cross_frame_precisions, cross_frame_recalls, cross_frame_f1 = [], [], []
+        final_tps, final_fps, final_fns, final_tns = [], [], [], []
+        frame_metrics_lst = []
+        cross_frame_tp_rate = []
+
+        for frame_id, match_lst in all_pedestrian_by_frames.items():
+            frame_metrics_match = compute_match_arrays(match_lst)
+            tp_count = len(frame_metrics_match.TP)
+            fp_count = len(frame_metrics_match.FP)
+            fns_count = len(frame_metrics_match.FN)
+            final_tps.extend(frame_metrics_match.TP)
+            final_fps.extend(frame_metrics_match.FP)
+            final_fns.extend(frame_metrics_match.FN)
+            final_tns.extend(frame_metrics_match.TN)
+
+            frame_precision = precision(tp_count, fp_count)
+            frame_recall = recall(tp_count, fns_count)
+            frame_f1 = f1_score(tp_count, fp_count, fns_count)
+            cross_frame_precisions.append(frame_precision)
+            cross_frame_recalls.append(frame_recall)
+            cross_frame_f1.append(frame_f1)
+            cross_frame_tp_rate.append(tp_count / len(row.high_risk))
+            frame_metrics_lst.append(
+                {
+                    "frame_id": frame_id,
+                    "FrameLevelPrecision": frame_precision,
+                    "FrameLevelRecall": frame_recall,
+                    "FrameLevelF1": frame_f1,
+                },
+            )
+
+        results.append(
+            {
+                "raw_inferences": df_raw_inferences,
+                "frame_metrics": frame_metrics_lst,
+                "TP": final_tps,
+                "FP": final_fps,
+                "FN": final_fns,
+                "TN": final_tns,
+                "has_TP": True if len(final_tps) > 0 else False,
+                "has_FP": True if len(final_fps) > 0 else False,
+                "has_FN": True if len(final_fns) > 0 else False,
+                "has_TN": True if len(final_tns) > 0 else False,
+                "CrossFramePrecision": np.mean(cross_frame_precisions),
```

Comment:
Just curious - where did these "cross frame" metrics come from? Is this a canonical method in object tracking evaluation, or did we find some issue with the usual object detection metrics that would be derived from TP/FP/FN above?

Reply:
These metrics came up in past discussions with @marko-polo-cheno as the most valuable metrics to highlight. The idea that the mean of precision, recall and f1 across all frames within a video is more meaningful than taking the total TP/FP/FN and then calculating precision, recall and f1.